### PR TITLE
Upgrade version to Spring Cloud Azure 6.0.0-beta3

### PR DESCRIPTION
**Pull request by moarychan**
## Purpose

Sync update issue https://github.com/Azure/azure-sdk-for-java/issues/31330 fix .

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

**Comment by moarychan** [comment]
Related issue: https://github.com/Azure/azure-sdk-for-java/issues/31330

**Comment by moarychan** [comment]
@saragluna please help review this PR, thanks!

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplication.java [hunk 1]
```diff
@@ -22,16 +26,13 @@ public static void main(String[] args) {
         SpringApplication.run(EventHubBinderApplication.class, args);
     }
 
-    // Replace destination with spring.cloud.stream.bindings.consume-in-0.destination
-    // Replace group with spring.cloud.stream.bindings.consume-in-0.group
-    @ServiceActivator(inputChannel = "{destination}.{group}.errors")
-    public void consumerError(Message<?> message) {
-        LOGGER.error("Handling consumer ERROR: " + message);
+    @Bean
+    public Consumer<MessageHandlingException> consumerError() {
```
Comment (saragluna) [inline eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplication.java:30]: Should we use the ErrorMessage here?

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplication.java [hunk 2]
```diff
@@ -22,16 +26,13 @@ public static void main(String[] args) {
...
     }
 
-    // Replace destination with spring.cloud.stream.bindings.supply-out-0.destination
-    @ServiceActivator(inputChannel = "{destination}.errors")
-    public void producerError(Message<?> message) {
-        LOGGER.error("Handling Producer ERROR: " + message);
+    @ServiceActivator(inputChannel = "errorChannel")
+    public void producerError(ErrorMessage errorMessage) {
+        LOGGER.error("Handling Producer ERROR", errorMessage);
     }
```
Comment (saragluna) [inline eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplication.java:37]: We can still use the errorChannel?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplication.java (hunk 1 above)
Comment (moarychan) [reply eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplication.java:30]: Yes, will update it.

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplication.java (hunk 2 above)
Comment (moarychan) [reply eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplication.java:37]: Yes, our IT covers this use-case https://github.com/Azure/azure-sdk-for-java/blob/feature/spring-boot-3/sdk/spring/spring-cloud-azure-integration-tests/src/test/java/com/azure/spring/cloud/integration/tests/eventhubs/binder/EventHubsBinderProduceErrorIT.java#L66

**Review by saragluna — DISMISSED** [review]
LGTM.

**Review by Netyyyy — APPROVED** [review]

**Pull request merged**